UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.SubscriptionService/SubscriptionServiceJob.cs

```diff
@@ -47,18 +47,41 @@ public async Task ProcessQueueMessage([QueueTrigger("%SubscriptionStorageConfigu
                 }
                 catch (Exception e)
                 {
-                    subscriptionRequestResult.ProvisioningState = e.Message;
-                    _logger.LogError(EventIds.CreateSubscriptionRequestError.ToEventId(),
-                   "Subscription provisioning request failed with Exception:{e} with SubscriptionId:{SubscriptionId} and _D365-Correlation-ID:{correlationId} and _X-Correlation-ID:{CorrelationId}", e.Message, subscriptionRequestResult.SubscriptionId, subscriptionMessage.D365CorrelationId, subscriptionMessage.CorrelationId);
+                    subscriptionRequestResult.ProvisioningState = "Failed";
+
+                    externalNotificationEntity.ResponseStatusCode = _d365CallbackConfiguration.Value.FailedStatusCode;
+
+                    //Webhook validation handshake failure error
+                    if (e.Message.Contains("Webhook validation handshake failed"))
+                    {
+                        int startIndex = e.Message.IndexOf("Webhook validation handshake failed");
+                        subscriptionRequestResult.ErrorMessage = e.Message.Substring(startIndex, e.Message.Length - startIndex);
+                        externalNotificationEntity.ResponseDetails = "Failed to add subscription @Time: " + Convert.ToString(DateTime.UtcNow) + " with exception " + subscriptionRequestResult.ErrorMessage;
+
+                        _logger.LogError(EventIds.CreateSubscriptionRequestHandshakeFailureError.ToEventId(),
+                  "Subscription provisioning request failed with Webhook handshake failure error with Exception:{e} with SubscriptionId:{SubscriptionId} and _D365-Correlation-ID:{correlationId} and _X-Correlation-ID:{CorrelationId}", e.Message, subscriptionRequestResult.SubscriptionId, subscriptionMessage.D365CorrelationId, subscriptionMessage.CorrelationId);
+                    }
+                    //other potential errors
+                    else
+                    {
+                        subscriptionRequestResult.ErrorMessage = e.Message;                        
+                        externalNotificationEntity.ResponseDetails = "Failed to add subscription @Time: " + Convert.ToString(DateTime.UtcNow) + " with exception " + e.Message;
+
+                        _logger.LogError(EventIds.CreateSubscriptionRequestPotentialError.ToEventId(),
+                  "Subscription provisioning request failed with other potential errors with Exception:{e} with SubscriptionId:{SubscriptionId} and _D365-Correlation-ID:{correlationId} and _X-Correlation-ID:{CorrelationId}", e.Message, subscriptionRequestResult.SubscriptionId, subscriptionMessage.D365CorrelationId, subscriptionMessage.CorrelationId);
```

Comment:
Not a potential error surely, it's an actual error so slight reword please:
```suggestion
                  "Subscription provisioning request failed with other error with Exception:{e} for SubscriptionId:{SubscriptionId} and _D365-Correlation-ID:{correlationId} and _X-Correlation-ID:{CorrelationId}", e.Message, subscriptionRequestResult.SubscriptionId, subscriptionMessage.D365CorrelationId, subscriptionMessage.CorrelationId);
```

Reply:
Modifed , Thanks